Refactored examples folder

mparkhe opened the pull request:
Used guidance offered on #314 

- Organized this folder as "hello world" and "tutorials". Partial tree 
```
examples/
├── README.md
├── hello_world
│   ├── hello_world.py
│   └── sklearn_logistic_regression.py
└── tutorials
    ├── backend_stores/...
    ├── h2o/...
    ├── multistep_workflow/...
    ├── pytorch/...
    ├── sklearn_elasticnet
    │   ├── train_diabetes.py
    │   └── wine_quality/...
    └── tensorflow/...
```
- Removed Jupyter notebooks from examples. Rewritten as python examples.
- Added `README.md` 

fixed #314

Commit:
refactor examples directory

Commit:
readme for examples

mateiz commented:
This is a good start but I think it's still too complex. Just have a single `examples` with one subdirectory in it for each example. In addition, there should be _one_ thing that people start with. For example, it's confusing whether you should start with `hello_world` or `tutorials`.

mparkhe commented:
I removed the other `example` folder (which had a couple of empty subfolders).

and renamed `hello_world` to `quick_start`. In addition, I split this into two:

```
examples/
├── quick_start
│   └── mlflow_tracking.py
└── quick_start_ml
     └── sklearn_logistic_regression.py
```

The motivation to club all the other under `tutorials` folder was to clean up the top level. The other alternative is to create a single subfolder for each example directly under `examples` folder.

```
examples/
├── quick_start
├── quick_start_ml
├── tutorial-h2o
├── tutorial-multistep_workflow
├── tutorial-pytorch
├── tutorial-remote_store
├── tutorial-resources
├── tutorial-sklearn_elasticnet_diabetes
├── tutorial-sklearn_elasticnet_wine
└── tutorial-tensorflow
```
I added this `tutorial-` prefix to qualify each of these examples and set them apart from the quick start example.

@mateiz : If you prefer the latter, i can do that.

andrewmchen commented:
Perhaps I missed it: Did we update the documentation to reflect the new paths of all the examples? Also, the notebooks might be included by design to demonstrate MLflow's usage in a notebook. We should check with @dennyglee who added it. https://github.com/mlflow/mlflow/pull/129/files.

mparkhe commented:
Thanks @andrewmchen -- will update documentation and also check with Denny.

mateiz commented:
I don't think we need `tutorial-` in front of the other ones. Just call them `h2o`, `tensorflow`, etc to make them easy to find. Also, I think we should have just one quick start, not two. Call the other one scikit-learn for example. Try to make it in sync with what we have at https://www.mlflow.org/docs/latest/index.html: it calls one of them quick start and one thing tutorial.

Commit:
flatten structure. bringing back the notebooks

Commit:
updated documentation

Commit:
Merge branch 'master' of github.com:databricks/mlflow into examples

dennyglee reviewed (approved):
LGTM

mateiz commented on examples/h2o/random_forest.py:
```diff
@@ -0,0 +1,32 @@
+test = wine[0.3 <= r]
+
+
+def trainRandomForest(ntrees):
```
Nit: I'd change this to train_random_forest in order to follow the naming convention in the rest of the code. I guess this is based on the notebook somehow.

mparkhe replied:
done

mateiz commented on examples/h2o/wine-quality.csv:
```diff
@@ -0,0 +1 @@
+../resources/wine-quality.csv
```
I think it might be better if each example had its own copy of the file so that it's self-contained, instead of putting the file in "resources". Otherwise if you just copy an example folder (a natural way to start your own project), it will suddenly stop working.

mparkhe replied:
done

Commit:
addressing comments

mparkhe commented on examples/h2o/random_forest.ipynb:
```diff
@@ -71,7 +71,7 @@
      },
      "metadata": {},
-     "output_type": "display_data"
+     "output_type": "execute_result"
```
`git revert` seemed to muck with this. I am reverting to old stage.

mparkhe commented on docs/source/quickstart.rst:
```diff
@@ -25,8 +25,8 @@ can leverage MLflow in your daily workflow.
-and cd into the ``example`` subdirectory of the repository. We'll use this working directory for
-running the quickstart.
+and cd into the ``examples`` subdirectory of the repository. We'll use this working directory for
+running the ``quick_start``.
```
done

Commit:
reverting notebook changes

mateiz commented:
LGTM, thanks for fixing it up!
